storage/innobase/fsp/fsp0fsp.cc

```diff
@@ -2092,7 +2104,7 @@ fseg_alloc_free_page_low(
 			first = flst_get_first(seg_inode + FSEG_FREE);
 		} else {
 			ut_ad(!has_done_reservation);
-			return(NULL);
+			return nullptr;
 		}
 
 		ret_descr = xdes_lst_get_descriptor(*space, first, mtr, &xdes);
```

Comment:
After the code that follows this, we are testing for `ret_page == FIL_NULL` twice. I think that we’d better avoid `goto alloc_done` if the allocation fails. Simply `return nullptr` in those cases.